includes/sanitizers/class-amp-story-export-sanitizer.php

```diff
@@ -0,0 +1,168 @@
+<?php
+/**
+ * Class AMP_Story_Export_Sanitizer.
+ *
+ * @package AMP
+ */
+
+/**
+ * Class AMP_Story_Export_Sanitizer
+ *
+ * Sanitizes AMP Stories during export.
+ *
+ * @since 1.2.1
+ */
+class AMP_Story_Export_Sanitizer extends AMP_Base_Sanitizer {
+
+	/**
+	 * Default args.
+	 *
+	 * @var array
+	 */
+	protected $DEFAULT_ARGS = [
+		'base_url' => '',
+	];
+
+	/**
+	 * Default assets.
+	 *
+	 * @var array
+	 */
+	protected $assets = [];
+
+	/**
+	 * Sanitize the AMP Story.
+	 */
+	public function sanitize() {
+
+		// AMP Story Node.
+		$this->sanitize_assets(
+			'amp-story',
+			[
+				'publisher-logo-src',
+				'publisher',
+				'poster-portrait-src',
+				'poster-square-src',
+				'poster-landscape-src',
+			]
+		);
+
+		// AMP Image Nodes.
+		$this->sanitize_assets(
+			'amp-img',
+			[
+				'src',
+				'srcset',
+			]
+		);
+
+		// AMP Video Nodes.
+		$this->sanitize_assets(
+			'amp-video',
+			[
+				'src',
+				'poster',
+			]
+		);
+
+		// Send the raw HTTP header for the export assets.
+		if ( ! empty( $this->assets ) && ! headers_sent() ) {
+			header( 'X-AMP-Export-Assets: ' . implode( ',', $this->assets ) );
+		}
+	}
+
+	/**
+	 * Add and sanitize the export assets for the AMP Story.
+	 *
+	 * @param string $name       The DOMElement name.
+	 * @param array  $attributes The DOMElement attributes.
+	 */
+	public function sanitize_assets( $name, $attributes ) {
+		$nodes     = $this->dom->getElementsByTagName( $name );
+		$num_nodes = $nodes->length;
+
+		// Verify we have a value to update the paths with.
+		$update_path = ! empty( $this->args['base_url'] ) ? $this->args['base_url'] : false;
+
+		/**
+		 * Generates the new asset path.
+		 *
+		 * @param string $asset The original URL.
+		 *
+		 * @return bool|string Returns false when $update_path is false, else the new URL.
+		 */
+		$get_asset_path = function( $asset ) use ( $update_path ) {
+			if ( $asset && $update_path ) {
+				$args = [
+					$update_path,
+					'assets',
+					basename( $asset ),
```

Comment:
What happens if there are two files uploaded (or linked to) with the same filename? I'm thinking there could be one at like `/wp-content/uploads/2018/12/christmas.jpg` and another at `/wp-content/uploads/2019/12/christmas.jpg`. So perhaps there should be a some some logging of which asset URLs have been processed and what their filenames are. If a new asset URL is processed and it would end up getting the same filename as a previous, then it should get some auto-increment added to the filename.

Reply:
@westonruter Perhaps instead of trying to track and ensure all the images are unique we convert `/wp-content/uploads/2019/12/christmas.jpg` to `2019-12-christmas.jpg` when moving it to the assets directory. This would avoid collisions.

Reply:
Replacing all slashes with hyphens is a great idea.